src/data_provider/tests/sysInfoPackageLinuxParserRpm/sysInfoPackageLinuxParserRPM_test.cpp

```diff
@@ -0,0 +1,504 @@
+
+    getRpmInfo([&wrapper](nlohmann::json & packageInfo)
+    {
+        wrapper.callbackMock(packageInfo);
+    });
+
+}
+
+
+TEST(SysInfoPackageLinuxParserRPM_test, rpmFromLibRPM)
+{
+    CallbackMock wrapper;
+
+    auto expectedPackage1 = R"({"name":"1","architecture":"2","description":"3","size":4,"version":"5:7-6","vendor":"8","install_time":"9","groups":"10","format":"rpm"})"_json;
+
+    auto utils_mock { std::make_unique<UtilsMock>() };
+    auto rpm_mock { std::make_unique<RpmLibMock>() };
+
+    m_utils_mock = utils_mock.get();
+    m_rpm_mock = rpm_mock.get();
+    rpmts ts = (rpmts) 0x123;
+    rpmtd td = (rpmtd) 0x123;
+    rpmdbMatchIterator mi = (rpmdbMatchIterator) 0x123;
+    Header header = (Header) 0x123;
+
+    EXPECT_CALL(*utils_mock, existsRegular(_)).Times(1).WillOnce(Return(false));
+    EXPECT_CALL(*rpm_mock, rpmReadConfigFiles(_, _)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(*rpm_mock, rpmtsCreate()).Times(1).WillOnce(Return(ts));
+
+
+    EXPECT_CALL(*rpm_mock, rpmtsOpenDB(_, _)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(*rpm_mock, rpmtsRun(_, _, _)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(*rpm_mock, rpmtdNew()).Times(1).WillOnce(Return(td));
+    EXPECT_CALL(*rpm_mock, rpmtsInitIterator(_, _, _, _)).Times(1).WillOnce(Return(mi));
+    EXPECT_CALL(*rpm_mock, rpmdbNextIterator(_)).WillOnce(Return(header)).WillOnce(Return(nullptr));
+    EXPECT_CALL(*rpm_mock, rpmtsCloseDB(_)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(*rpm_mock, rpmtsFree(_)).Times(1).WillOnce(Return(nullptr));
+    EXPECT_CALL(*rpm_mock, rpmtdFree(_)).Times(1).WillOnce(Return(nullptr));
+    EXPECT_CALL(*rpm_mock, rpmdbFreeIterator(_)).Times(1).WillOnce(Return(nullptr));
+    EXPECT_CALL(*rpm_mock, rpmFreeRpmrc());
+
+    EXPECT_CALL(*rpm_mock, headerGet(_, _, _, _)).Times(AnyNumber()).WillRepeatedly(Return(1));
+
+    EXPECT_CALL(*rpm_mock, rpmtdGetString(_)) \
+    .WillOnce(Return("1")) \
+    .WillOnce(Return("7")) \
+    .WillOnce(Return("6")) \
+    .WillOnce(Return("summary")) \
+    .WillOnce(Return("8")) \
+    .WillOnce(Return("10")) \
+    .WillOnce(Return("source")) \
+    .WillOnce(Return("2")) \
+    .WillOnce(Return("3"));
+    EXPECT_CALL(*rpm_mock, rpmtdGetNumber(_)).WillOnce(Return(5)).WillOnce(Return(9)).WillOnce(Return(4));
+
+
+    EXPECT_CALL(wrapper, callbackMock(expectedPackage1)).Times(1);
+
+    getRpmInfo([&wrapper](nlohmann::json & data)
+    {
+        wrapper.callbackMock(data);
+    });
+
+}
+
+TEST(SysInfoPackageLinuxParserRPM_test, rpmFallbackFromLibRPM)
+{
+    CallbackMock wrapper;
+
+    auto expectedPackage1 = R"({"name":"1","architecture":"2","description":"3","size":4,"version":"5:7-6","vendor":"8","install_time":"9","groups":"10","format":"rpm"})"_json;
+    auto expectedPackage2 = R"({"name":"11","architecture":"12","description":"13","size":14,"version":"15:17-16","vendor":"18","install_time":"19","groups":"20","format":"rpm"})"_json;
```

Comment:
Idem of the comment above

Reply:
This is true at first sight, but applying such change be have an impact on callback prototypes
```
std::function<void(nlohmann::json&)> callback
```